### DISCUSSION TITLE
How to disable all major and minor upgrade for stable

### DISCUSSION THREAD

**Discussion by georgkrause** [renovatebot/renovate#13652] [Request Help] [answered]
I have the following scenario: The project has two branches, one is called `develop` where actual code changes happen. I want to have all dependencies updated there. Another branch, `stable`, has the latest minor version and we do patch releases from there. Now I want to achieve that the stable branch don't get dependency updates with breaking changes, so I'd like to configure renovate to only do lock file maintenance and patch releases there. 

As far as I understand the documentation, the `packageRules` conditions are only able to be combined with `OR`, so I am not sure how to achieve what I want. Am I missing something? Some pointers would be really cool! 

Thank you all in advance :)

**Comment by viceice** [comment]
https://docs.renovatebot.com/configuration-options/#matchbasebranches ?

Most `match...` operators are `and` nor `or`.

**Reply by georgkrause** [reply]
I am aware of `matchBaseBranches`, but I don't know how to proceed. Can I add `matchUpdateTypes`, and simply disable for the combination of `stable` + major/minor?

**Reply by rarkins** [reply]
How would you solve your stable requirement *without* worrying about base branches?

**Comment by georgkrause** [comment, marked as answer]
It turns out, this works:

```
{
  "baseBranches": ["stable", "develop"],
  "packageRules": [
    {
      "matchUpdateTypes": ["major", "minor"],
      "matchBaseBranches": ["stable"],
      "enabled": false
    }
  ]
}
```

What really confused me is the following quote from the docs:

> Note how the above uses matchPackageNames instead of matchPackagePatterns because it is an exact match package name. This is the equivalent of defining "matchPackagePatterns": ["^aws\-sdk$"] and hence much simpler. However you can mix together both matchPackageNames and matchPackagePatterns in the same package rule and the rule will be applied if either match. 

I am aware that this is only for the specific rule, but as far as I read the docs there is no information about the logic combination of different rules. Maybe this would be a good thing to add?

Anyway, thanks for your help everyone!

**Reply by rarkins** [reply]
Great, thanks for confirming. I was confused earlier and didn't realize that "stable" meant the branch. I thought you meant only for stable releases
